### PR TITLE
front: fixed bug or first number of the position field cannot be cleared

### LINKED ISSUE #7866: Infra editor : First number of Position field cannot be erased
### What happened?

I try to split a track section in the infrastructure editor, and I want to change the position of the split : I want to erase the position (2500) and write a new one (5000) : I cannot erase the "2"

image

I try to erase everything : 

image

### What did you expect to happen?

I expected to be able to fully erase my number because I cannot change from 2500 to 5000
But it is important that one cannot validate the split with no value 

### How can we reproduce it (as minimally and precisely as possible)?

As above

### What operating system, browser and environment are you using?

- **Browser**: Chrome
- **OS**: W11
- **Env**: Staging internal

### OSRD version (top right corner `Account` button > `Informations`)

d9655c0

### PR DESCRIPTION
**Pull request by theocrsb**
close #7866

### CODE REVIEW

**Review by Uriel-Sautron — APPROVED** [review]
LGTM !!! Tested

File: front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx [hunk 1]
```diff
@@ -86,7 +86,7 @@ const TrackSplitLeftPanel: FC<unknown> = () => {
             id="offset"
             type="number"
             inputProps={{ required: true }}
-            value={state.offset / 1000}
+            value={state.offset === 0 ? '' : state.offset / 1000}
```
Comment (anisometropie) [inline front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx:89]: If we forbid them to write 0 we need to change the tooltip as well 
Screenshot_20240628_113436

File: front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx [hunk 2]
```diff
@@ -100,8 +100,10 @@ const TrackSplitLeftPanel: FC<unknown> = () => {
                 : undefined
             }
             onChange={(e) => {
-              if (!Number.isNaN(e.target.valueAsNumber))
-                setState((prev) => ({ ...prev, offset: e.target.valueAsNumber * 1000 }));
+              setState((prev) => ({
+                ...prev,
+                offset: (!Number.isNaN(e.target.valueAsNumber) ? e.target.valueAsNumber : 0) * 1000,
```
Comment (anisometropie) [inline front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx]:
```suggestion
                offset: e.target.valueAsNumber * 1000 || 0,
```

**Review by anisometropie — CHANGES_REQUESTED** [review]

**Review by kmer2016 — COMMENTED** [review]
I've reviewed/tested  the changes and agree with the previously made comments. Once those adjustments are made, I believe the PR will be ready to merge. Good job!

File: front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx (hunk 2 above)
Comment (theocrsb) [reply front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx]: I think we need to check if e.target.valueAsNumber is a number for the use case where the input is empty

File: front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx (hunk 1 above)
Comment (theocrsb) [reply front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx:89]: After validation of a PO we do not accept the value 0

File: front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx (hunk 2 above)
Comment (kmer2016) [inline front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx]: As discussed with @theocrsb, the value can be NaN if the user enters nothing. For improved readability, you can write:
```suggestion
                offset: (e.target.valueAsNumber ?? 0) * 1000,
```
We use the [nullish coalesing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)

File: front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx (hunk 1 above)
Comment (theocrsb) [reply front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx:89]: Done

File: front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx (hunk 2 above)
Comment (kmer2016) [reply front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx]: Never mind nullish coalesing is not applyed on NaN
https://github.com/tc39/proposal-nullish-coalescing/issues/28

**Review by kmer2016 — APPROVED** [review]

File: front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx (hunk 2 above)
Comment (anisometropie) [reply front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx]: !Number.isNaN is not a really good test, it says true for anything that is not `NaN`
Screenshot_20240628_163719

Comment (anisometropie) [reply front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx]: `offset: e.target.valueAsNumber * 1000` is going to be a number only if `e.target.valueAsNumber` is a number

Comment (theocrsb) [reply front/src/applications/editor/tools/trackSplit/components/TrackSplitLeftPanel.tsx]: Ok thanks for the explanations! Done

**Comment by clarani** [comment]
Can you please squash your commits ? Thanks :)

**Review by anisometropie — APPROVED** [review]

**Pull request merged**